### PR TITLE
[WIP] Fix for queries stuck in queue and resource group concurrency not respected by coordinator

### PR DESCRIPTION
**Pull request by swapsmagic**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java [hunk 1]
```diff
@@ -181,7 +212,7 @@ public void testTwoQueriesAtSameTime()
         waitForQueryState(queryRunner, secondDashboardQuery, QUEUED);
     }
 
-    @Test(timeOut = 90_000)
+    //@Test(timeOut = 90_000)
```
Comment (tdcmeehan) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java:215]: Typo

File: presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStateProvider.java [hunk 2]
```diff
@@ -355,10 +445,12 @@ private static BasicQueryInfo createQueryInfo(String queryId, QueryState state,
 
     private static BasicQueryInfo createQueryInfo(String queryId, QueryState state, String resourceGroupId, MemoryPoolId memoryPool, DataSize totalMemoryReservation)
     {
+        String[] splits = resourceGroupId.split("\\.");
+        List<String> resourceGroupList = Arrays.asList(splits);
         return new BasicQueryInfo(
                 new QueryId(queryId),
                 TEST_SESSION.toSessionRepresentation(),
-                Optional.of(new ResourceGroupId(resourceGroupId)),
+                Optional.of(new ResourceGroupId(Arrays.asList(splits))),
```
Comment (tdcmeehan) [inline presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStateProvider.java]: I would add a helper method on `ResourceGroupId` that removes the boilerplate of splitting and recreating an id.

**Review by tdcmeehan — APPROVED** [review]

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java (hunk 1 above)
Comment (swapsmagic) [reply presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java]: removed

File: presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStateProvider.java (hunk 2 above)
Comment (swapsmagic) [reply presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStateProvider.java]: done as part of the constructor itself.

**Pull request closed without merging**